vault/identity_store_util.go

```diff
@@ -144,7 +146,7 @@ func (i *IdentityStore) activateDeduplication(ctx context.Context, req *logical.
 		return fmt.Errorf("failed to reset existing identity state: %w", err)
 	}
 
-	if err := i.loadArtifacts(ctx); err != nil {
+	if err := i.loadArtifacts(ctx, i.localNode.HAState() == consts.Active); err != nil {
```

Comment:
Is this going to work for all nodes? I thought this function was supposed to be called after activation in all nodes, including non-active ones.

Reply:
It should, but only the active node will be persisting, which I think this should handle! Hopefully I'm understanding your question correctly. Please let me know if there's something I'm missing.

Reply:
Ahhh right!